velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -73,6 +73,31 @@ Aws::IOStreamFactory AwsWriteableStreamFactory(void* data, int64_t nbytes) {
   return [=]() { return Aws::New<StringViewStream>("", data, nbytes); };
 }
 
+folly::Synchronized<
+    std::unordered_map<std::string, AWSCredentialsProviderFactory>>&
+credentialsProviderFactories() {
+  static folly::Synchronized<
+      std::unordered_map<std::string, AWSCredentialsProviderFactory>>
+      factories;
+  return factories;
+}
+
+std::optional<std::shared_ptr<Aws::Auth::AWSCredentialsProvider>>
+getCredentialsProviderByName(
+    const std::string& providerName,
+    const S3Config& s3Config) {
+  return credentialsProviderFactories().withRLock(
+      [&](const auto& factories)
+          -> std::optional<std::shared_ptr<Aws::Auth::AWSCredentialsProvider>> {
+        const auto it = factories.find(providerName);
+        if (it == factories.end()) {
```

Comment:
We should probably error out here, similar to other registrations.

Reply:
This was to address comment https://github.com/facebookincubator/velox/pull/12774#discussion_r2010486998 Do you mean we should change the behaviour back to throwing an exception?

Reply:
That comment was for the test only. We don't have to use fmt. We can print the exact message.
Yes, throwing an exception clarifies that the configuration was incorrect. Missing registration in this case.

Reply:
I misunderstood that comment. Thanks for the explanation!